test/DateTimeFormatterTest.php

```diff
@@ -45,98 +27,100 @@ public static function returnUnfilteredDataProvider(): array
                 ],
             ],
             [0.53],
+            ['2024-31-31'],
+            [true],
+            [false],
         ];
     }
 
-    #[DataProvider('returnUnfilteredDataProvider')]
-    public function testReturnUnfiltered(mixed $input): void
+    #[DataProvider('returnExceptionDataProvider')]
+    public function testReturnExceptionForUnfilteredValues(mixed $input): void
     {
-        date_default_timezone_set('UTC');
-
         $filter = new DateTimeFormatter();
 
-        self::assertSame($input, $filter($input));
+        $this->expectException(Exception\InvalidArgumentException::class);
+
+        $filter->filter($input);
     }
 
     public function testFormatterFormatsZero(): void
     {
-        date_default_timezone_set('UTC');
-
         $filter = new DateTimeFormatter();
         $result = $filter->filter(0);
-        self::assertSame('1970-01-01T00:00:00+0000', $result);
+        self::assertSame('1970-01-01T00:00:00+00:00', $result);
     }
 
     public function testDateTimeFormatted(): void
     {
-        date_default_timezone_set('UTC');
-
         $filter = new DateTimeFormatter();
```

Comment:
The timezone should be explicitly set here so that tests don't fail on user machines in different time zones